From aa200247cfa2516ead950cbb2312fe869c6d1a6f Mon Sep 17 00:00:00 2001
From: Yaoming Qin <b10945@freescale.com>
Date: Wed, 26 Jan 2011 15:51:00 +0800
Subject: [PATCH 116/486] ENGR00138392 A strip of screen flashes when switch
 to camcorder mode.

 The root cause in commit fef065e15a0a2a7c4d is not correct. The root
 cause is that the initial position of output is 0, which is not handled
 correctly.

 Signed-off-by: Yaoming Qin<B10945@freescale.com>
---
 liboverlay/overlay.cpp      |  6 ------
 liboverlay/overlay_thread.h | 10 ++++++++--
 2 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/liboverlay/overlay.cpp b/liboverlay/overlay.cpp
index 303bff1..21c28c9 100755
--- a/liboverlay/overlay.cpp
+++ b/liboverlay/overlay.cpp
@@ -839,12 +839,6 @@ static int overlay_init_v4l(struct overlay_control_context_t *dev)
                 return -1;
             }
 
-			//reset the v4l input buffer
-			for (int i = 0; i < (dev->xres * dev->yres); i ++){
-				vir_addr[2*i] = 0;
-				vir_addr[2*i + 1] = 0x80;
-				}
-
             dev->v4lbuf_addr[buf_index] = vir_addr;
             //v4l already init this buffer to black
             //fill the v4l to black;
diff --git a/liboverlay/overlay_thread.h b/liboverlay/overlay_thread.h
index 9ccab14..ede8b3d 100755
--- a/liboverlay/overlay_thread.h
+++ b/liboverlay/overlay_thread.h
@@ -383,7 +383,10 @@ class OverlayThread: public Thread {
             }
 
             //Mixer the first buffer from overlay instance0 to V4L2 Buffer
-            if(overlay_buf0) {
+            if(overlay_buf0 && (overlayObj0->outX != 0  ||
+                     overlayObj0->outY != 0  ||
+                     overlayObj0->outW != 0  ||
+                     overlayObj0->outH != 0)) {
                 //Setting input format
                 mIPUInputParam.width = overlayObj0->mHandle.width;
                 mIPUInputParam.height = overlayObj0->mHandle.height;
@@ -456,7 +459,10 @@ class OverlayThread: public Thread {
 
             //Check whether we need to do another mixer, based on
             //buffers in overlay instance1's buffer queue
-            if(overlay_buf1) {
+            if(overlay_buf1 && (overlayObj1->outX != 0  ||
+                     overlayObj1->outY != 0  ||
+                     overlayObj1->outW != 0  ||
+                     overlayObj1->outH != 0)) {
                 //Setting input format
                 mIPUInputParam.width = overlayObj1->mHandle.width;
                 mIPUInputParam.height = overlayObj1->mHandle.height;
-- 
1.8.0

